# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PBrowser.ipdl
# Commit: 0df2f2832755
# Full Hash: 0df2f283275530fbfe5cf9c95986b073c0d6a31e
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 1: Use BigBuffer for IPCDataTransfer, r=edgar
#   
#   The IPCDataTransfer type is used to transfer Clipboard/Drag & Drop payloads
#   over IPC to allow them to be written to or read from the relevant system
#   interfaces. Previously, the system which was used was somewhat complex, and
# ==============================================================================

diff -r 1d21d911b3e7 -r 0df2f2832755 dom/ipc/PBrowser.ipdl
--- a/dom/ipc/PBrowser.ipdl	Mon Aug 22 15:35:28 2022 +0000
+++ b/dom/ipc/PBrowser.ipdl	Mon Aug 22 15:38:14 2022 +0000
@@ -107,6 +107,7 @@
 using mozilla::AspectRatio from "mozilla/AspectRatio.h";
 using mozilla::NativeKeyBindingsType from "mozilla/NativeKeyBindingsType.h";
 using mozilla::StyleImageRendering from "mozilla/ServoStyleConsts.h";
+[MoveOnly] using class mozilla::ipc::BigBuffer from "mozilla/ipc/BigBuffer.h";
 
 namespace mozilla {
 namespace dom {
@@ -415,7 +416,7 @@
      */
     async SetCursor(nsCursor value,
                     bool hasCustomCursor,
-                    nsCString customCursorData,
+                    BigBuffer? customCursorData,
                     uint32_t width, uint32_t height,
                     float resolutionX, float resolutionY,
                     uint32_t stride, SurfaceFormat format,